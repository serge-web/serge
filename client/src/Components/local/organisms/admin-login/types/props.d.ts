export default interface Props {
  /**
   * Handle Enter button click event
   */
  onSubmit: (password: string) => void
}
