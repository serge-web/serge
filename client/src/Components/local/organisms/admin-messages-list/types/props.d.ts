import { MessageChannel } from 'src/custom-types'

export default interface PropTypes {
  /**
   * The messages array
   */
  messages: Array<MessageChannel>
  /*
   *  The function to perform to mark all messages as read
   */
  markAllAsRead: any
  /**
   * Current force that is seeing the messages
   */
  force: {
    name: string
    color: string
    icon?: string
  }
}
