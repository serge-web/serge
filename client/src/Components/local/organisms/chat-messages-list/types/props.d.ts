import { ChatMessage, MessageInfoTypeClipped } from 'src/custom-types'
import { TurnFormats } from 'src/config'
import ForcesInChannelProps from '../../../molecules/forces-in-channel/types/props'

export default interface PropTypes extends ForcesInChannelProps {
  /**
   * The list of channel messages properties required
   * for ChatMessage components. This will either
   * be chat messages or info messages (used for game turn marker)
   */
  messages: Array<ChatMessage | MessageInfoTypeClipped>
  isUmpire: boolean
  /** force this player belongs to */
  playerForce: string
  /** role this player belongs to */
  playerRole: string
  /**
   * Max height of chat container box
   */
  chatContainerHeight?: number

  turnPresentation?: TurnFormats

  observing?: boolean
  markUnread?: (message: ChatMessage) => void
  /** whether to hide the forces block for this channel */
  hideForcesInChannel: boolean
  /** whether to hide the author (force and role) of messages */
  hideAuthor?: boolean
}
