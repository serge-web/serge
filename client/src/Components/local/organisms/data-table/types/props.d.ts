import React from 'react'
import { ROW_DATA_TYPE, ROW_WITH_COLLAPSIBLE_TYPE } from '..'

export interface RowDataType {
  type: typeof ROW_DATA_TYPE
  rowkey: string
  component: React.ReactElement | null
  label: string
}

export interface RowWithCollapsibleType {
  type: typeof ROW_WITH_COLLAPSIBLE_TYPE
  rowKey: string
  // collapsible is a function witch one will return a ReactElement and will be rendered only on call
  // cb: is a callback function with one should fire some logick on collapsible content change, ex close collapsible component
  collapsible: (cb?: () => void) => React.ReactElement
  cells: Array<RowDataType>
}

export type Column =
  | string
  | {
      filters: Array<string>
      label: string
    }
export type RowType = RowDataType | RowWithCollapsibleType

export default interface Props {
  /**
   * List of columns
   */
  columns: Column[]
  /**
   * List of data
   */
  data: RowType[]

  /** toggle table sort */
  sort?: boolean

  /** no display expand icon */
  noExpand?: boolean
}
