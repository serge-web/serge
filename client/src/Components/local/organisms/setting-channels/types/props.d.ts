import { ChannelTypes, ForceData, TemplateBody } from 'src/custom-types'
import { Item } from 'src/local/molecules/editable-list'
import { Option } from 'src/local/molecules/editable-row'

export default interface PropTypes {
  channels: Array<ChannelTypes>
  messageTemplates: Array<TemplateBody>
  forces: Array<ForceData>
  onChange: (updates: { channels: Array<ChannelTypes>, selectedChannel: ChannelTypes }) => void
  onSave?: (channel: ChannelTypes) => void
  onSidebarClick?: (selectedChannel: ChannelTypes) => void
  onCreate?: (name: string, channel: ChannelTypes) => void
  onDelete?: (item: ChannelTypes) => void
  onDuplicate?: (item: Item) => void
  selectedChannel?: ChannelTypes
}

export type MessageGroupProps = {
  title: string
  multiple: boolean
  options?: Option[]
  onChange: (value: string[]) => void
  onDelete?: (value: string[]) => void
  type: string
  value: string[]
}

export { Channel, ChannelCollab, ChannelTypes, ForceData, Participant, Role } from 'src/custom-types'
