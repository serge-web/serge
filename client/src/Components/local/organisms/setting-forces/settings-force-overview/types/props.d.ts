import { ForceData } from 'src/custom-types'

export default interface PropTypes {
  data: ForceData
  handleChangeForce: (obj: ForceData) => void
}
