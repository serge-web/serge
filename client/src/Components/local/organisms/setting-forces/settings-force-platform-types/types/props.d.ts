import { ForceData, Route, PlatformTypeData, Asset } from 'src/custom-types'

export default interface PropTypes {
  selectedForce: ForceData
  platformTypes: PlatformTypeData[]
  onChangeHandler: (nextForce: ForceData) => void
  routes?: Route[]
  forcesData: ForceData[]
  onDeleteAsset?: () => void
  selectedAssetItem?: Asset
}
