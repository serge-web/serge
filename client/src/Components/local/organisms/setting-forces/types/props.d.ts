import { ForceData } from 'src/custom-types'
import { Item } from '../../molecules/sortable-list'

type Forces = Array<ForceData>

export default interface PropTypes {
  forces: Forces
  onChange: (forces: { forces: Array<ForceData> }) => void
  onSave: (forces: Array<ForceData>) => void
  onRejectedIcon?: (rejected: any) => void
  onSidebarClick?: (selectedForce: ForceData) => void
  onCreate?: () => void
  onDelete?: (item: ForceData) => void
  onDuplicate?: (item: ForceData) => void
  selectedForce?: ForceData
  iconUploadUrl?: string
  /** Handler for when user tries to delete role with Game Control privileges */
  customDeleteHandler?: (NewItems: Item[], key: number, handleChange: (changedItems: Item[]) => void) => void
}

export { ForceData, Role } from 'src/custom-types'
