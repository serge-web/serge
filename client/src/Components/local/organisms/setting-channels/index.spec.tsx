/* global it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import SettingChannels from './index'
import { ChannelTypes } from './types/props'
import { WargameExportedMock, MessageTemplatesMock } from 'src/mocks'

const onChange = (updates: { channels: Array<ChannelTypes> }): void => { console.log(updates.channels) }

describe('SettingChannels component:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SettingChannels
        channels={WargameExportedMock.data.channels.channels || []}
        forces={WargameExportedMock.data.forces.forces}
        onChange={onChange}
        messageTemplates={MessageTemplatesMock}
      />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
