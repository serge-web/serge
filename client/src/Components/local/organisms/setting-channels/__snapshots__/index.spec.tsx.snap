// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingChannels component: renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="left"
  >
    <div
      className="actions"
    >
      <div
        aria-label="split button"
        className="MuiButtonGroup-root MuiButtonGroup-contained"
        role="group"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            ADD CHANNEL
          </span>
        </button>
        <button
          aria-haspopup="menu"
          aria-label="select merge strategy"
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="main"
    >
      <div
        className="searchlist"
      >
        <section
          className="input-container disable-offset fullwidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <div
              className="MuiInputBase-root MuiInput-root makeStyles-root-4 makeStyles-root-6 MuiInput-underline makeStyles-underline-5 makeStyles-underline-7 MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name=""
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search"
                required={false}
                type="text"
                value=""
              />
            </div>
          </div>
        </section>
        <div
          className="searchlist-list searchlist-list"
        >
          <span
            className="searchlist-row "
            onClick={[Function]}
          >
            <span>
              Channel 16
            </span>
            <div
              className="searchlist-menu"
            >
              <svg
                className="svg-inline--fa fa-trash"
              />
            </div>
          </span>
          <span
            className="searchlist-row "
            onClick={[Function]}
          >
            <span>
              Blue Net
            </span>
            <div
              className="searchlist-menu"
            >
              <svg
                className="svg-inline--fa fa-trash"
              />
            </div>
          </span>
          <span
            className="searchlist-row "
            onClick={[Function]}
          >
            <span>
              Mapping
            </span>
            <div
              className="searchlist-menu"
            >
              <svg
                className="svg-inline--fa fa-trash"
              />
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="right"
  >
    <div
      className="row"
    >
      <div
        className="col"
      />
      <div
        className="actions"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary Mui-disabled Mui-disabled"
          data-qa-type="save"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Save Channel
          </span>
        </button>
      </div>
    </div>
    <div>
      Channels empty. Please create a channel.
    </div>
  </div>
</div>
`;
